lib/haskell/natural4/src/LS/XPile/CoreL4/LogicProgram/Common.hs

```diff
@@ -98,51 +98,4 @@ data OpposesClause t = OpposesClause
   }
   deriving (Eq, Generic, Ord, Read, Show)
 
-instance Hashable t => Hashable (OpposesClause t)
-
--- Derive Generic and Hashable instances for some Babyl4 types.
-deriving instance Generic (QVarName t)
-deriving instance Generic (Var t)
-deriving instance Generic Val
-deriving instance Generic ClassName
-deriving instance Generic (Tp t)
-deriving instance Generic (VarDecl t)
-
-deriving instance Generic UArithOp
-deriving instance Generic UBoolOp
-deriving instance Generic UTemporalOp
-deriving instance Generic UnaOp
-
-deriving instance Generic BArithOp
-deriving instance Generic BBoolOp
-deriving instance Generic BComparOp
-deriving instance Generic BinOp
-
-deriving instance Generic Quantif
-deriving instance Generic FieldName
-deriving instance Generic ListOp
-
-deriving instance Generic (Expr t)
-
-instance Hashable t => Hashable (QVarName t)
-instance Hashable t => Hashable (Var t)
-instance Hashable Val
-instance Hashable ClassName
-instance Hashable t => Hashable (Tp t)
-instance Hashable t => Hashable (VarDecl t)
-
-instance Hashable UArithOp
-instance Hashable UBoolOp
-instance Hashable UTemporalOp
-instance Hashable UnaOp
-
-instance Hashable BArithOp
-instance Hashable BBoolOp
-instance Hashable BComparOp
-instance Hashable BinOp
-
-instance Hashable Quantif
-instance Hashable FieldName
-instance Hashable ListOp
-
-instance Hashable t => Hashable (Expr t)
\ No newline at end of file
+instance Hashable t => Hashable (OpposesClause t)
```

Comment:
No need to derive all the other Generic and Hashable instances anymore, because that is now done in babyl4 itself.